mythril/laser/ethereum/instructions.py

```diff
@@ -1184,18 +1186,30 @@ def extcodecopy_(self, global_state: GlobalState) -> List[GlobalState]:
             global_state=global_state,
         )
 
-    @StateTransition
+    @StateTransition()
     def extcodehash_(self, global_state: GlobalState) -> List[GlobalState]:
         """
 
         :param global_state:
         :return: List of global states possible, list of size 1 in this case
         """
-        # TODO: To be implemented
-        address = global_state.mstate.stack.pop()
-        global_state.mstate.stack.append(
-            global_state.new_bitvec("extcodehash_{}".format(str(address)), 256)
-        )
+        world_state = global_state.world_state
+        stack = global_state.mstate.stack
+        address = Extract(159, 0, stack.pop())
+
+        if address.symbolic:
+            log.debug("unsupported symbolic address for EXTCODEHASH")
+            stack.append(global_state.new_bitvec("extcodehash_" + str(address), 256))
```

Comment:
Hmm... I don't like this, because it is completely symbolic. If, for example code had:
```
require(extcodehash(addr) == 1),
```
this would think it is possible.

I think returning the code hash of an address with no code would be better here, as in general, we've been treating symbolic addresses as non-contract (user) addresses, which still have a codehash, but its always the same.